src/lib.rs

```diff
@@ -88,6 +88,23 @@ pub use handler::*;
 ///  - For `Enum::MyVariant(param1, ..., paramN)` and `Enum::MyVariant { param1,
 ///    ..., paramN }`, the payload is `(param1, ..., paramN)` (where `N`>1).
 ///
+/// You can also go with even more complexity and write a syntax similar to rusts pattern matching
+/// system:
+///
+///  - For `Enum::MyVariant(SecondEnum::MyVariant)` and
+///    `Enum::MyVariant { param: SecondEnum::MyVariant }`, the payload is `SecondEnum::MyVariant`
+///  - For `Enum::MyVariant { param1, param2: SomeStruct { param3 } }`, the payload is `(param1,
+///    param3)`
+///  - For `Enum::MyVariant { param1, .. }` and `Enum::MyVariant( param1, .. )`, the payload is
+///    `(param1,)`
+///  - For `Enum::MyVariant { param1: Some(param2) }` and `Enum::MyVariant( Some(param2) )`, the payload is
+///    `param2`
+///  - For `Enum::MyVariant { param1: None::<T> }` and `Enum::MyVariant( None::<T> )`, the payload is
+///    `Option<T>` that is equal to None::<T>
+///
+/// It's recursive, so on every parameter you can insert a new struct, tuple struct, enum variant or
+/// anything else.
+///
```

Comment:
In the examples below, we should add at least one (simple) example demonstrating the new functionality you're adding.